Libplanet/Blockchain/Policies/BlockPolicy.cs

```diff
@@ -131,9 +131,20 @@ DateTimeOffset currentTime
         /// <inheritdoc />
         public int GetNextBlockDifficulty(IEnumerable<Block<T>> blocks)
         {
-            return ExpectDifficulties(blocks, yieldNext: true)
-                .First(t => t.Block == null)
-                .Difficulty;
+            int blockCount = blocks.Count();
+
+            if (blockCount <= 1)
+            {
+                return blockCount;
+            }
+
+            Block<T>[] prevBlocks = blocks.Skip(Math.Max(0, blockCount - 2))
+                .ToArray();
+
+            return GetNextDifficultyFromPrevTimestamp(
+                prevBlocks[0].Timestamp,
+                prevBlocks[1].Timestamp,
+                prevBlocks[1].Difficulty);
         }
 
         private IEnumerable<DifficultyExpectation> ExpectDifficulties(
```

Comment:
Does this function still used?

Reply:
Right, this is still used in `ValidateBlocks()`. I'll find a way to get rid of `ExpectDifficulties()`.